cmd/cosign/cli/options/attest.go

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package options
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// AttestOptions is the top level wrapper for the attest command.
+type AttestOptions struct {
+	Key       string
+	Cert      string
+	Upload    bool
+	Force     bool
+	Recursive bool
+
+	Fulcio       FulcioOptions
+	SecurityKey  SecurityKeyOptions
+	Predicate    PredicateOptions
+	RegistryOpts RegistryOpts
+}
+
+// AddAttestOptions adds the sign command options to cmd.
+func AddAttestOptions(cmd *cobra.Command, o *AttestOptions) {
```

Comment:
Rather than having out-param `*AttestOptions`, why not have it be a method on `*AttestOptions`, and have an interface commands can implement?

e.g. https://github.com/mattmoor/mink/blob/2d3b8ed868dccebb68553f7dbb4979cd9933b3ca/pkg/command/bundle.go#L82-L83

Reply:
I like that, I can start moving to that pattern.

Reply:
https://github.com/sigstore/cosign/pull/791